zilliqa/src/p2p_node.rs

```diff
@@ -234,8 +234,14 @@ impl P2pNode {
 
         if let Some((peer, address)) = &self.config.bootstrap_address {
             if self.swarm.local_peer_id() != peer {
+                // Add bootstrap node's address and call Kademlia bootstrap
+                self.swarm
+                    .behaviour_mut()
+                    .kademlia
+                    .add_address(peer, address.clone());
+                self.swarm.behaviour_mut().kademlia.bootstrap()?;
+
                 self.swarm.dial(address.clone())?;
-                self.swarm.add_peer_address(*peer, address.clone());
```

Comment:
Why don't we add the peer address to the swarm? I'm not sure, but I guess this would also let other behaviours (e.g. gossipsub) learn about the bootstrap peer?

Reply:
I removed it because we're explicitly adding to Kademlia above. I suppose youre right it doesnt hurt to do both for the other protocols